Arrowgene.Ddon.Shared/AssetRepository.cs

```diff
@@ -126,16 +120,46 @@ private void RegisterAsset<T>(Action<T> onLoadAction, string key, IAssetDeserial
         private void Load<T>(Action<T> onLoadAction, string key, IAssetDeserializer<T> readerWriter)
         {
             string path = Path.Combine(_directory.FullName, key);
-            FileInfo file = new FileInfo(path);
-            if (!file.Exists)
+
+            FileSystemInfo info = File.GetAttributes(path).HasFlag(FileAttributes.Directory) ? new DirectoryInfo(path) : new FileInfo(path);
+            if (!info.Exists)
             {
-                Logger.Error($"Could not load '{key}', file does not exist");
+                Logger.Error($"The file or directory '{key}' does not exist");
+                return;
             }
 
             try {
-                T asset = readerWriter.ReadPath(file.FullName);
-                onLoadAction.Invoke(asset);
-                OnAssetChanged(key, asset);
+                if (info is DirectoryInfo)
+                {
+                    // Special handeling for quests
+                    if (readerWriter is QuestAssetDeserializer)
```

Comment:
Why the special case?

Reply:
Well, I could not figure out how to pass in the same asset multiple times as our current pattern has it always return the asset created. Here we are trying to collect for the same asset from parsing multiple files. Suggestions?

Reply:
Maybe the AssetDeserializers for directories should take the directory instead of the file, and it'd be understood that asset deserializers for directories would know how to collect them all, or maybe the deserializer would be called for each file, it'll return the loaded assets in each file, and this method would collect them? 🤔

Reply:
Originally I was trying to do this, but imagine now that directory has 10000000 files. 1 file has changed. Do you want to reload all of them? or just 1?

Reply:
For that you'd need to have something set up to track which element of the collection corresponds to which file. This should maybe be invisible to the AssetRepository? Maybe a DirectoryAssetDeserializer class, which the QuestAssetDeserializer would subclass should be the one in charge of doing it

Reply:
This file watcher does exactly that I think though? That is what it looked like when I was messing around with it. After the initial load, it would only trigger if I made a new file or modified an existing one, for that file only. At least, that is what I thought was happening.

Reply:
Yeah, it's probably not worth it to hot reaload in general, but quests especially, would be more trouble than its worth it. I think it can be merged without hot reloading.

Reply:
Still, it should not require a special case like this 🤔

Reply:
I removed the special case for Quests. There still need to be a check either for directory or file as we need to enumerate the files in the directory.

Reply:
Fine by me, having a special case for directories/files is a reasonable compromise